### DISCUSSION TITLE
[Tempo] Show isolated services in service graphs visualization

### DISCUSSION THREAD

**Discussion by mapno** [grafana/grafana#64379] [Data Sources]
[Service graph metrics](https://grafana.com/docs/tempo/latest/metrics-generator/service-graph-view/) are only generated when a service communicates with one or more services in the same system. That leaves out of the graph many services that don't match the requirements to derive service graph metrics:
* Instrumentation doesn't match the requirements for service graphs—parent/children spans tagged with `span.kind`
* Services that don't communicate with other services, for any reason.

For these cases, we could use spanmetrics, as we do for the APM table, and show them if they're not present in service graph metrics.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82797. Feel free to continue the discussion around this there. Thank you!
